bin/meli-test-account.js

```diff
@@ -0,0 +1,147 @@
+#!/usr/bin/env node
+require('dotenv').config({path: require('path').resolve(process.cwd(), '.env.test')})
+require('../config/db').connect()
+
+const app = require('express')()
+const req = require('request-promise')
+const chromeLauncher = require('chrome-launcher')
+const {Spinner} = require('cli-spinner')
+
+const auth = require('../routes/auth')
+const Account = require('../model/account')
+const {refresh} = require('../config/meli-auth')
+
+const devAccountUsername = process.env.DEV_ACCOUNT_USERNAME
+const port = process.env.PORT
+const TIMEOUT_MS = 60 * 1000
+
+const global = {
+  spinner: null
+}
+
+async function getDevAccount() {
+  const account = await Account.findOne({nickname: devAccountUsername})
+  if (!account) {
+    throw new Error(`Dev Account username ${devAccountUsername} not found.`)
+  }
+  if (!account.isAuthorized()) {
+    const accessToken = await refresh.requestNewAccessToken('mercadolibre', account.auth.refreshToken)
+    await account.updateAccessToken(accessToken)
+  }
+  return account
+}
+
+async function requestTestAccount(devAccount) {
+  const testAccountRequestOptions = {
+    method: 'POST',
+    uri: 'https://api.mercadolibre.com/users/test_user',
+    qs: {
+      access_token: devAccount.auth.accessToken
+    },
+    body: {
+      site_id: 'MLA'
+    },
+    json: true // Automatically stringifies the body to JSON
+  }
+  return req(testAccountRequestOptions)
+}
+
+function onAuthSuccess() {
+  global.spinner.stop(true)
+  console.log('Test account Register success! Exiting...')
+  process.exit()
+}
+
+function onAuthAbort() {
+  global.spinner.stop(true)
+  console.log('Chrome login window closed.')
+  process.exit()
+}
+
+async function launchChrome(loginUrl) {
+  console.log(`Waiting for login on: ${loginUrl}`)
+  const chrome = await chromeLauncher.launch({
+    startingUrl: loginUrl
+  })
+  chrome.process.once('exit', onAuthAbort)
+  console.log(`Chrome window opened.`)
+  return chrome
+}
+
+function startWaitLoginSpinner() {
+  const dateStart = new Date()
+  const dateEnd = new Date(dateStart.getTime() + TIMEOUT_MS)
+  const spinner = new Spinner({
+    text: `%s `,
+    stream: process.stderr,
+    onTick(msg) {
+      const now = new Date()
+      const millisLeft = dateEnd.getTime() - now.getTime()
```

Comment:
The only thing you are using of `dateEnd` is the `getTime()` function, why you don't just make dateEnd save the time you need and avoid creating a `new Date()` object? 
e.g. 
```js
// variable declaration
const dateEnd = dateStart.getTime() + TIMEOUT_MS

// usage
const millisLeft = dateEnd - now.getTime()
```
We save an object and also we help the poor gargage collector :)

Reply:
Good suggestion, didn't see that one! I'll do it